Realm/ObjectServerTests/SwiftFlexibleSyncServerTests.swift

```diff
@@ -1315,6 +1356,337 @@ extension SwiftFlexibleSyncServerTests {
+
+        var results0 = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe()
+        XCTAssertEqual(results0.count, 3)
+        XCTAssertEqual(realm.subscriptions.count, 1)
+        results0 = try await results0.where { $0.age < 8 }.subscribe() // results0 local query is { $0.age >= 8 AND $0.age < 8 }
+        XCTAssertEqual(results0.count, 0) // no matches because local query is impossible
+        XCTAssertEqual(realm.subscriptions.count, 2) // two subscriptions: "$0.age >= 8 AND $0.age < 8" and "$0.age >= 8"
+        let results1 = realm.objects(SwiftPerson.self)
+        XCTAssertEqual(results1.count, 3) // three objects from "$0.age >= 8". None "$0.age >= 8 AND $0.age < 8".
+    }
+
+    @MainActor
+    func testSubscribeSameQueryNoName() async throws {
+        try await populateSwiftPerson()
+        let realm = try openFlexibleSyncRealm()
+
+        let results0 = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe()
+        let ex = XCTestExpectation(description: "no attempt to re-create subscription, returns immediately")
+        realm.syncSession!.suspend()
+        Task {
+            _ = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe()
+            _ = try await results0.subscribe()
+            XCTAssertEqual(realm.subscriptions.count, 1)
+            ex.fulfill()
+        }
+        await fulfillment(of: [ex], timeout: 2.0)
+        XCTAssertEqual(realm.subscriptions.count, 1)
+    }
+
+    @MainActor
+    func testSubscribeSameQuerySameName() async throws {
+        try await populateSwiftPerson()
+        let realm = try openFlexibleSyncRealm()
+
+        let results0 = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe(name: "8 or older")
+        realm.syncSession!.suspend()
+        let ex = XCTestExpectation(description: "no attempt to re-create subscription, returns immediately")
+        Task {
+            _ = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe(name: "8 or older")
+            _ = try await results0.subscribe(name: "8 or older")
+            XCTAssertEqual(realm.subscriptions.count, 1)
+            ex.fulfill()
+        }
+        await fulfillment(of: [ex], timeout: 5.0)
+        XCTAssertEqual(realm.subscriptions.count, 1)
+    }
+
+    @MainActor
+    func testSubscribeSameQueryDifferentName() async throws {
+        try await populateSwiftPerson()
+        let realm = try openFlexibleSyncRealm()
+
+        let results0 = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe()
+        _ = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe(name: "8 or older")
+        _ = try await results0.subscribe(name: "older than 7")
+        XCTAssertEqual(realm.subscriptions.count, 3)
+        let first = realm.subscriptions.first(ofType: SwiftPerson.self) { $0.age >= 8 }
+        XCTAssertNil(first?.name)
```

Comment:
This is asserting that either `first` or `first.name` is `nil`, and would pass if the subscription was missing entirely. `XCTAssertNil(try XCTUnwrap(first).name)` is a more precise test.